src/dataService/savedItemsService.ts

```diff
@@ -422,6 +429,214 @@ export class SavedItemDataService {
     }
     // Tags are a many-to-one relationship with item, so need
     // to take distinct results after performing this join
-    return this.readDb.select('*').from(baseQuery.as('base')).distinct();
+    return this.db.select('*').from(baseQuery.as('base')).distinct();
+  }
+
+  /**
+   * Format a date to the configured database timezone
+   * @param date
+   * @private
+   */
+  private static formatDate(date: Date): string {
+    return mysqlTimeString(date, config.database.tz);
+  }
+
+  /**
+   * Update the 'favorite' attribute of an item, and the auditing fields
+   * in the table ('time_updated', etc.)
+   * @param itemId the item ID to update
+   * @param favorite whether the item is a favorite or not
+   */
+  public async updateSavedItemFavoriteProperty(
+    itemId: string,
+    favorite: boolean
+  ): Promise<void> {
+    this.db = this.writeDb;
```

Comment:
any functions associated with the mutation call will switch the default db to `writeDb`...so any following getSavedItems call made with `this` object will only use write client

Reply:
what about updating the query methods to optionally take a database connection instead (and default to the read connection in the class). I tend to try to avoid manipulating class properties within functions because it's harder to debug/see what's going on, in my opinion.

Reply:
That way we can just explicitly use the write connection in the mutation resolvers

Reply:
ya thats another way of implementing this and we thought about that during the discussion..then at resolver layer, we need to pass the write client instead of read with every call and we want to hide those details from the resolver layer (and let it handle only business logic)

Reply:
Ok, I would prefer the other way but if you and Kelvin decided on it then I will support

Reply:
I just really dislike class state

Reply:
My alternative suggestion is that the update method should return the data required by the resolver. The only reason it's separated is because the classes weren't merged together. However that is no longer the case.

Reply:
had a discussion with kat and kelvin: https://pocket.slack.com/archives/C034FB364LC/p1645815901665619

changed the resolver to pass the databaseClient object to service, and also modified the upsert/update functions to return the mutation return object.